# Source data and code
Regarding the article "Cross-species dissection of saline-related genes by genetically deciphering a euryhaline microalga Chlorella sp.”. We have provided complete and detailed source data and executable code.

# GWAS analysis of MEM25
This document outlines a comprehensive ​​GWAS (Genome-Wide Association Study) analysis pipeline​​ for the MEM25​​, focusing on variant detection, quality control, and association testing with phenotypic traits. The workflow begins with raw sequencing data preprocessing (e.g., Fastp filtering, BWA alignment, GATK-based variant calling) and emphasizes rigorous quality checks, including duplicate removal, base quality recalibration (BQSR) skipping (due to MEM25’s non-model status), and hard filtering of SNPs/INDELs based on metrics like QD, FS, and MQ. Key steps include GVCF generation, joint genotyping, population-level PCA to assess stratification (minimal in MEM25 mutants), and phenotype normalization to mitigate batch effects. The final GWAS employs PLINK for linear regression, using kinship matrices and PCA covariates to link genetic variants (MAF ≥ 0.05) to growth-related traits (dry weight, cell size). The pipeline highlights MEM25’s industrial potential while addressing challenges like low heterozygosity (0.00743) and sequencing depth trade-offs (e.g., ~35X to 25X post-Q30 filtering). Results are validated through statistical thresholds and visualizations (e.g., PCA plots, density distributions), ensuring robust variant-phenotype associations.
